### PR TITLE
feat: Port Alert as ToastAlert and animation polish

### PR DESCRIPTION
**Pull request by ling1726**
Ports the `Alert` component from the `@fluentui/react-alert` package as `ToastAlert` (for now) and adds visual polish.

Also adds visual polish to animations in the Toast and Toaster
components to be able to support variable sized Toasts

### CODE REVIEW

File: packages/react-components/react-toast/src/components/Alert/Alert.tsx [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+import { useAlertStyles_unstable } from './useAlertStyles.styles';
+
+import type { AlertProps } from './Alert.types';
+import type { ForwardRefComponent } from '@fluentui/react-utilities';
+
+/**
+ * An Alert component displays a brief, important message to attract a user's attention
+ *  without interrupting their current task.
+ */
+export const Alert: ForwardRefComponent<AlertProps> = React.forwardRef((props, ref) => {
```
Comment (ling1726) [inline packages/react-components/react-toast/src/components/Alert/Alert.tsx:14]: will port the stories and documentation in a follow up PR, this package is still private so there's no rush

File: packages/react-components/react-toast/src/components/Toast.tsx [hunk 2]
```diff
@@ -95,9 +91,27 @@ export const Toast: React.FC<ToastProps & { visible: boolean }> = props => {
     }
   }, [play, toastRef]);
 
+  const onEntering = () => {
+    if (!toastRef.current) {
+      return;
+    }
+
+    const element = toastRef.current;
+    element.style.setProperty('--fui-toast-height', `${element.scrollHeight}px`);
```
Comment (ling1726) [inline packages/react-components/react-toast/src/components/Toast.tsx:100]: it's impossible to animate to `height: auto`, so using some JS magic and transitions to help here

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/dc612644bc698fa8a8f315167d759f16c9a41d82'>dc612644bc698fa8a8f315167d759f16c9a41d82</a></sub>
<div id="bundleSizeComment9423"/>

**Pull request merged**